python/src/aac/plugins/constraint_assignment_arguments/constraint_assignment_arguments_impl.py

```diff
@@ -0,0 +1,171 @@
+"""The AaC Constraint assignment arguments plugin implementation module."""
+# NOTE: It is safe to edit this file.
+# This file is only initially generated by aac gen-plugin, and it won't be overwritten if the file already exists.
+
+# There may be some unused imports depending on the definition of the plugin...but that's ok
+from aac.execute.aac_execution_result import (
+    ExecutionResult,
+    ExecutionStatus,
+    ExecutionMessage,
+    MessageLevel,
+)
+from aac.context.language_context import LanguageContext
+from aac.context.definition import Definition
+from typing import Any
+
+
+plugin_name = "Constraint assignment arguments"
+
+
+def check_arguments_against_constraint_definition(  # noqa: C901
+    instance: Any, definition: Definition, defining_schema
+) -> ExecutionResult:
+    """Business logic for the Check arguments against constraint definition constraint."""
+
+    context = LanguageContext()
+    if not context.is_aac_instance(
+        instance, "aac.lang.SchemaConstraintAssignment"
+    ) and not context.is_aac_instance(
+        instance, "aac.lang.PrimitiveConstraintAssignment"
+    ):
+        # the constraint failed
+        error_msg = ExecutionMessage(
+            f"The Check arguments against constraint definition constraint for {instance.name} failed because the instance is not a SchemaConstraintAssignment or PrimitiveConstraintAssignment.  You may only use this constraint on SchemaConstraintAssignment or PrimitiveConstraintAssignment definitions.  Received {type(instance)}.)",
+            MessageLevel.ERROR,
+            definition.source,
+            None,
+        )
+        return ExecutionResult(
+            plugin_name,
+            "Check arguments against constraint definition",
```

Comment:
The fun of looking between multiple plugins and starting to follow the logic better. So, I was incorrect, the original text is the command of the plugin, `check` is the higher command that the constraint is rolled up into for executing. 

Going to leave this thread here in case someone else hits a similar snag but remove the repeated change suggestions.